### PR TITLE
added change log and test

### PR DESCRIPTION
**Pull request by nicola-giuliani**
just a minor log and a test added to complete #12293

### CODE REVIEW

File: tests/grid/tria_clear_user_data.cc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "../tests.h"
+
+template <int dim>
+void
+test()
+{
+  Triangulation<dim> tria;
+  GridGenerator::hyper_cube(tria);
+  deallog << 'Cleared user data in ' << dim << 'D' << std::endl;
```
Comment (bangerth) [inline tests/grid/tria_clear_user_data.cc:31]: Don't you actually want to call the function you are testing?

File: tests/grid/tria_clear_user_data.output [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+DEAL::Cleared user data in 1D
+DEAL::Cleared user data in 2D
+DEAL::Cleared user data in 3D
```
Comment (bangerth) [inline tests/grid/tria_clear_user_data.output:4]: Please add an endline character to the end of this file, as well as to the changelog entry.

Comment (nicola-giuliani) [reply tests/grid/tria_clear_user_data.output]: done, thanks.

File: tests/grid/tria_clear_user_data.cc (hunk 1 above)
Comment (nicola-giuliani) [reply tests/grid/tria_clear_user_data.cc:32]: whops, forgot one line. thanks

File: doc/news/changes/minor/20210525NicolaGiuliani [hunk 3]
```diff
@@ -0,0 +1,3 @@
+New: Fixed a bug in clear user data for standard Triangulation.
+<br>
+(Nicola Giuliani, 2021/05/25)
```
Comment (bangerth) [inline doc/news/changes/minor/20210525NicolaGiuliani:3]: This file is still missing an endline at the end.

Comment (nicola-giuliani) [reply doc/news/changes/minor/20210525NicolaGiuliani]: sorry...

**Review by kronbichler — APPROVED** [review]

File: tests/grid/tria_clear_user_data.cc (hunk 1 above)
Comment (kronbichler) [reply tests/grid/tria_clear_user_data.cc:32]: To fix the error on the test system,
```
/jenkins/workspace/dealii_PR-12302/tests/grid/tria_clear_user_data.cc:32:14: error: character constant too long for its type [-Werror]
   deallog << 'Cleared user data in ' << dim << 'D' << std::endl;
```
please change this into
```suggestion
  deallog << "Cleared user data in " << dim << 'D' << std::endl;
```

**Review by kronbichler — APPROVED** [review]

**Pull request merged**